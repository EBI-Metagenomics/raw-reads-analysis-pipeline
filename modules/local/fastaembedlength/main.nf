process FASTAEMBEDLENGTH {
    tag "${meta.id}"
    label 'process_single'

    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/python:3.13'
        : 'quay.io/biocontainers/python:3.13'}"

    input:
    tuple val(meta), path(fasta)
    path script

    output:
    tuple val(meta), path("${out_fn}"), emit: fasta
    path "versions.yml"               , emit: versions

    script:
    def args = task.ext.args ?: ''
    def base = fasta.getName().tokenize('.')[0..-2].join('.')
    def ext = fasta.getName().tokenize('.')[-1]
    out_fn = "${base}.renamed.${ext}"

    """
    python ${script} -i ${fasta} -o ${out_fn} --output_gzip

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version |& sed '1!d ; s/python //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def base = fasta.getName().tokenize('.')[0..-2].join('.')
    def ext = fasta.getName().tokenize('.')[-1]
    out_fn = "${base}.renamed.${ext}"
    
    """
    touch ${out_fn}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version |& sed '1!d ; s/python //')
    END_VERSIONS
    """
}
