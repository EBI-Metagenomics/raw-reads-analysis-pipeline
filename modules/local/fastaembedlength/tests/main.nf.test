nextflow_process {

    name "Test Process FASTAEMBEDLENGTH"
    script "../main.nf"
    process "FASTAEMBEDLENGTH"

    tag "modules"
    tag "modules_"
    tag "fastaembedlength"

    test("mini sample fasta should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    file("${projectDir}/tests/fixtures/fastaembedlength/ERR10889147_mini_1.fa.gz", checkIfExists: true)
		]
		input[1] = file("${projectDir}/bin/fastx_embed_length.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("mini sample fastq should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    file("${projectDir}/tests/fixtures/fastaembedlength/ERR10889147_mini_1.fq.gz", checkIfExists: true)
		]
		input[1] = file("${projectDir}/bin/fastx_embed_length.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("synthetic sample fasta should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    file("${projectDir}/tests/fixtures/fastaembedlength/synthetic_test.fa.gz", checkIfExists: true)
		]
		input[1] = file("${projectDir}/bin/fastx_embed_length.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("synthetic sample fastq should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    file("${projectDir}/tests/fixtures/fastaembedlength/synthetic_test.fq.gz", checkIfExists: true)
		]
		input[1] = file("${projectDir}/bin/fastx_embed_length.py")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
