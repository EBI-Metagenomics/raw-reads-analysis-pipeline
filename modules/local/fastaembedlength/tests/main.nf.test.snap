{
    "synthetic sample fasta should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "synthetic_test.fa.renamed.gz:md5,a1440ea08ce60d24e8c3de3296c75ede"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "synthetic_test.fa.renamed.gz:md5,a1440ea08ce60d24e8c3de3296c75ede"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T12:39:15.553236392"
    },
    "mini sample fastq should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889147_mini_1.fq.renamed.gz:md5,0939ecbff76ea6b9efa28f33f9b257b2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889147_mini_1.fq.renamed.gz:md5,0939ecbff76ea6b9efa28f33f9b257b2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T12:38:51.329996888"
    },
    "synthetic sample fastq should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "synthetic_test.fq.renamed.gz:md5,fada8b0fa11d09275c49e001609d8abc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "synthetic_test.fq.renamed.gz:md5,fada8b0fa11d09275c49e001609d8abc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T12:43:13.123928712"
    },
    "mini sample fasta should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889147_mini_1.fa.renamed.gz:md5,26284b0cf239160dfaadb8627d9b3c8d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889147_mini_1.fa.renamed.gz:md5,26284b0cf239160dfaadb8627d9b3c8d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dbaa3527686224ee480498eb1132a30e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T12:37:49.538576093"
    }
}